contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,621 @@
+    /// @notice Info of each user that stakes LP tokens.
+    mapping(uint256 => mapping(address => UserInfo)) public userInfo;
+    /// @dev Total allocation points. Must be the sum of all allocation points in all pools.
+    uint256 public totalAllocPoint;
+    uint256 private constant ACC_TOKEN_PRECISION = 1e18;
+    /// @dev A maximum scaling factor applied to boosted amounts.
+    uint256 public maxBoostFactor = 1500;
+    /// @dev Amount of claimable Joe the user has, this is required as we
+    /// need to update rewardDebt after a token operation but we don't
+    /// want to send a reward at this point. This amount gets added onto
+    /// the pending amount when a user claims.
+    mapping(uint256 => mapping(address => uint256)) public claimableJoe;
+
+    event Add(uint256 indexed pid, uint256 allocPoint, IERC20 indexed lpToken, IRewarder indexed rewarder);
+    event Set(uint256 indexed pid, uint256 allocPoint, IRewarder indexed rewarder, bool overwrite);
+    event Deposit(address indexed user, uint256 indexed pid, uint256 amount);
+    event Withdraw(address indexed user, uint256 indexed pid, uint256 amount);
+    event UpdatePool(uint256 indexed pid, uint256 lastRewardTimestamp, uint256 lpSupply, uint256 accJoePerShare);
+    event Harvest(address indexed user, uint256 indexed pid, uint256 amount);
+    event EmergencyWithdraw(address indexed user, uint256 indexed pid, uint256 amount);
+    event Init();
+    event MaxBoostFactorChanged(uint256 maxBoostFactor);
+
+    /// @param _MASTER_CHEF_V2 The JoeSwap MCJV2 contract address.
+    /// @param _joe The JOE token contract address.
+    /// @param _MASTER_PID The pool ID of the dummy token on the base MCJV2 contract.
+    constructor(
+        IMasterChef _MASTER_CHEF_V2,
+        IERC20 _joe,
+        IERC20 _veJoe,
+        uint256 _MASTER_PID
+    ) public {
+        MASTER_CHEF_V2 = _MASTER_CHEF_V2;
+        JOE = _joe;
+        VEJOE = _veJoe;
+        MASTER_PID = _MASTER_PID;
+    }
+
+    /// @notice Deposits a dummy token to `MASTER_CHEF_V2` MCJV2. This is required because MCJV2 holds the minting rights for JOE.
+    /// Any balance of transaction sender in `dummyToken` is transferred.
+    /// The allocation point for the pool on MCJV2 is the total allocation point for all pools that receive double incentives.
+    /// @param dummyToken The address of the ERC-20 token to deposit into MCJV2.
+    function init(IERC20 dummyToken) external onlyOwner {
+        uint256 balance = dummyToken.balanceOf(msg.sender);
+        require(balance != 0, "MasterChefV2: Balance must exceed 0");
+        dummyToken.safeTransferFrom(msg.sender, address(this), balance);
+        dummyToken.approve(address(MASTER_CHEF_V2), balance);
+        MASTER_CHEF_V2.deposit(MASTER_PID, balance);
+        emit Init();
+    }
+
+    /// @notice Returns the number of BMCJ pools.
+    function poolLength() external view returns (uint256 pools) {
+        pools = poolInfo.length;
```

Comment:
nit: `numPools` might be a better naming

Reply:
I'm tempted to leave this as is, I greped through the codebase and their seems to be some convention of having a `poolLength` function, possibly used for frontend.